pkg/arvo/lib/ethio.hoon

```diff
@@ -126,28 +126,30 @@
   =/  m  (strand:strandio ,block)
   ^-  form:m
   |^
+  %+  (retry:strandio ,block)  `10
+  =/  m  (strand:strandio ,(unit block))
+  ^-  form:m
   ;<  =json  bind:m
     %+  request-rpc  url
     :-  `'block by number'
     [%eth-get-block-by-number number |]
-  =/  =block  (parse-block json)
-  ?.  =(number number.id.block)
-    (strand-fail:strandio %reorg-detected >number< >block< ~)
-  (pure:m block)
+  (pure:m (parse-block json))
   ::
   ++  parse-block
     |=  =json
-    ^-  block
-    =<  [[&1 &2] |2]
-    ^-  [@ @ @]
+    ^-  (unit block)
+    =<  ?~(. ~ `[[&1 &2] |2]:u)
+    ^-  (unit [@ @ @])
```

Comment:
Probably better to just slap some faces on here? The tree addressing here is nothing complex, but it's still tree addressing. (I'd also use `=;` or `=-` here, but no strong feelings.)